### PR TITLE
Issue587 shorted StrProp builder

### LINKED ISSUE #587: Shorted Property builder methods for StrProp
**Is your feature request related to a problem? Please describe.**
The Property builder validation methods are too long.  They all start with the word 'must', even though the meaning is very clear without it.  There are good examples in String, JUnit and Hamcrest for better names, so lets follow those examples.

For this task:
* Update the StrProp.StrBuilder methods to the new names
* Old methods should be deprecated (the intention is to remove them in the 0.5.0 release).  Javadoc deprecation should reference the new method.
* Old methods should delegate to the new methods to keep the need for new testing to a minimum
* All usages in the AndHow project should be updated to the new method names.
* The StrPropTest can get by with just a single test invocation of each old method.  The existing testing will move to the new methods.

(Its probably easiest to use an IDE to rename the methods first, then copy paste to the old names... or some other clever way)

The chosen names are based on thediscussion #605:

| Current Name | Proposed New Name | Notes |
| ----------------- | --------------------- | ------------------------- |
| mustMatchRegex | matches | Same as `String.matches(regex)` |
| mustStartWith      | startsWith | Same as `String.startsWith(prefix)` |
| mustStartWith      | startsWith | Same as `String.startsWith(prefix)` |
| mustStartWithIgnoreCase | startsWithIgnoringCase | based on Hamcrest Matchers |
| mustEndWith       | endsWith | Same as `String.endsWith(suffix)` (sufix is misspelled on this method) |
| mustEndWithIgnoreCase | endsWithIgnoringCase | based on Hamcrest Matchers |
| mustEqual(String...) | oneOf(String...) | based on Hamcrest Matchers |

### PR DESCRIPTION
**Pull request by alex-kar**
Fixes #587 

### All Submissions:

Shorted Property builder:
    * Update StrProp.StrBuilder methods to the new names.
    * Mark old methods as deprecated, delegate to new methods.
    * Add javadoc for deprecated methods with references to new methods.
    * Update all usages to the new method names.

Have you checked that...
* [x] Your pull request is to the *_main_* branch
* [x] Your code is up to date with the latest code from the *_main_*
* [x] You followed the guidelines in our [Developer Guide](https://sites.google.com/view/andhow/developer)?
* [x] Your code does not decrease test coverage (maybe it improves coverage!!)
* [x] If this is related to an issue, please include a link to the issue with the 'Fixes #XXX' syntax.

### CODE REVIEW

**Comment by alex-kar** [comment]
@eeverman The StrPropTest keeps using deprecated methods, but code coverage percentage is reduced a little bit (0.01%). Should I fix it?

File: andhow-core/src/main/java/org/yarnandtail/andhow/property/StrProp.java [hunk 1]
```diff
@@ -43,32 +43,80 @@ public StrProp build() {
...
+		/**
+		 * @deprecated Use {@code StrBuilder.endsWith()}
+		 */
+		@Deprecated
 		public StrBuilder mustEndWith(String sufix) {
-			this.validation(new StringValidator.EndsWith(sufix, false));
+			return this.endsWith(sufix);
+		}
+
+		public StrBuilder endsWith(String suffix) {
```
Comment (eeverman) [inline andhow-core/src/main/java/org/yarnandtail/andhow/property/StrProp.java:93]: Thanks for catching this spelling mistake!

**Comment by eeverman** [comment]
I think the issue with the test coverage is that there already was missing test coverage on one of the validation method (equals, which became oneOf).  This change adds another line of code that refers to it, so there is an added untested line.

I think its fine for now.  Looking at the Property tests in general, they need improvement.  I'll make a separate task.

**Review by eeverman — APPROVED** [review]
Nice work - Its a lot of files affected by this change, but I think you got everything!
